src/Datadog.Trace/Propagation/PropagationExtensions.cs

```diff
@@ -0,0 +1,36 @@
+using System.Collections.Generic;
+using Datadog.Trace.Headers;
+
+namespace Datadog.Trace.Propagation
+{
+    internal static class PropagationExtensions
+    {
+        public static void Inject(this IPropagator propagator, SpanContext context, IHeadersCollection headers)
+        {
+            propagator.Inject(context, headers, (carrier, header, value) => carrier.Set(header, value));
```

Comment:
The lambdas are simple to follow and read, but there are usually fewer allocations involved when you just use a static method because these lambdas are not capturing any closures.